Sources/MongoSwift/ChangeStream.swift

```diff
@@ -99,7 +107,8 @@ public class ChangeStream<T: Codable>: CursorProtocol {
             type: .tailableAwait
         )
         self.client = client
-        self.decoder = decoder
+        self.decoder = BSONDecoder(copies: decoder, options: nil)
```

Comment:
since we are updating the userInfo I figured it was a bit safer to make a copy.

this did make me think about the fact that right now users can't give us userInfo that we should use when decoding their custom types, even though they can configure all of the other encoder/decoder options via the strategies. I don't think that's too urgent though, we can wait until if/when someone asks for this capability.